docs: add misssing networkpolicy docs (#1434)
kyma-project/istio: docs/user/tutorials/01-50-send-requests-using-egress.md

   If successful, the logs contain the request made by the egress Gateway:
   ```
   {"requested_server_name":"kyma-project.io","upstream_cluster":"outbound|443||kyma-project.io",[...]}
   ```

## Enhance Security by Implementing NetworkPolicies

Istio by default cannot securely enforce that egress traffic goes through the egress gateway. It only enables the flow
through sidecar proxies.
Kubernetes Network Policies can be used to restrict namespace traffic, so it only goes through the istio egress gateway.
Network policies are the Kubernetes way to enforce traffic rules in the namespace.

> [!NOTE] Support for network policies depends on the kubernetes CNI plugin used in the cluster.
> Make sure to check the documentation of the CNI plugin you are using.

> > [!WARN] In Gardener-based clusters the Network Policy restricting DNS traffic may notwork as expected.
> > It is due to local DNS service used in discovery working outside the CNI. You may need to define the `ipBlocks` with
> > the IP CIDR of the `kube-dns` service to the NetworkPolicy to allow proper DNS resolution.

Reviewer: ```suggestion
> [!WARN] In Gardener-based clusters, such as SAP BTP, Kyma runtime, the Network Policy restricting DNS traffic may not work as expected.
> It is due to the local DNS service used in discovery working outside the CNI. In such cases, define the **ipBlocks** with
> the IP CIDR of the `kube-dns` service in the NetworkPolicy to allow proper DNS resolution.
```

> 1. In the `$NAMESPACE` namespace, create a NetworkPolicy that allows only egress traffic to the istio egress gateway,

Reviewer: ```suggestion
1. Fetch the IP address of the `kube-dns` service:
```

> 1. In the `$NAMESPACE` namespace, create a NetworkPolicy that allows only egress traffic to the istio egress gateway,
>    blocking all other egress traffic. Fetch the IP address of the `kube-dns` service:
>    ```bash
>     export KUBE_DNS_ADDRESS=$(kubectl get svc -n kube-system kube-dns -o jsonpath='{.spec.clusterIP}')
>    ```
>    Create the NetworkPolicy with fetched IP address in the `ipBlocks` section:

Reviewer: ```suggestion
1. Fetch the IP address of the `kube-dns` service:
   ```bash
    export KUBE_DNS_ADDRESS=$(kubectl get svc -n kube-system kube-dns -o jsonpath='{.spec.clusterIP}')
   ```
2. Create a NetworkPolicy with the fetched IP address in the **ipBlocks** section. The NetworkPolicy allows only egress traffic to the Istio egress gateway, blocking all other egress traffic.
```

> 2. Create the NetworkPolicy with fetched IP address in the `ipBlocks` section:

Reviewer: ```suggestion
2. Create a NetworkPolicy with the fetched IP address in the **ipBlocks** section. The NetworkPolicy allows only egress traffic to the Istio egress gateway, blocking all other egress traffic.
```
